### ISSUE TITLE
NA in Relative abundance graph

### ISSUE THREAD

**Issue by AnnaMarizzi** [ChiLiubio/microeco#389]
Hello,

when I do a graph on the relative abundance for the genus of my data, NA appears on the x axis. 

Is there a function to eliminate the NA from the graph? Or do I have to filter my data differently?

Thank you in advance :)

![Problem](https://github.com/user-attachments/assets/22a3524c-3f8a-496d-b8f1-82c3ac2d4038)

**Comment by ChiLiubio** [comment]
Hi. I guess the reason is the taxonomic table is not tidied. Try to run this:
```
library(magrittr)
your_microtable$tax_table %<>% tidy_taxonomy
your_microtable$tidy_dataset()
your_microtable$cal_abund()
```
Then re-perform your analysis.

**Comment by AnnaMarizzi** [comment]
Thank you very much, it has solved the problem. Can you explain shortly what this function does. I have seen that it doesnt remove rows from the tax_table. From what I have read it replaces NA with " ", is that correct? And I guess it does that at any taxonomic level?

Should I run this always at the begining when I do the data filtering?

Thank you,
Anna

**Comment by ChiLiubio** [comment]
Yes. This function can replace all the so-called useless thing (NA, unknown, unclassified, ......) with "" and add the prefix (e.g. p__) for each taxonomic level. The reason we create this function is the taxonomic table we often got is chotic. I have seen many wired things in that. So unifying the table generally is necessary. So add it into your pipeline (before creating microtable), just once. It is enough.

Best,
Chi
